### PR TITLE
Add function to broadcast limited-area HEALPix to full shape

### PR DESCRIPTION
**Pull request by lkluft**

### CODE REVIEW

File: easygems/healpix/__init__.py [hunk 1]
```diff
@@ -43,6 +43,14 @@ def isel_extent(dx, extent):
     return np.arange(get_npix(dx))[get_extent_mask(dx, extent)]
 
 
+def broadcast_array(dx, fill_value=np.nan):
+    """Broadcast a limited-area HEALPix array to full shape."""
+    arr = fill_value * np.ones(shape=get_npix(dx))
```
Comment (d70-t) [inline easygems/healpix/__init__.py:48]: This sounds like `np.full` would do a better job, and maybe you want to keep the `dtype`.
```suggestion
    arr = np.full(get_npix(dx), fill_value, dtype=dx.dtype)
```

File: easygems/healpix/__init__.py [hunk 2]
```diff
@@ -43,6 +43,14 @@ def isel_extent(dx, extent):
...
+    arr[dx.cell] = dx.values
```
Comment (d70-t) [inline easygems/healpix/__init__.py:49]: Maybe you'd want to replace actually both of the lines with something based on either `xarray.DataArray.where`, or the three-argument version of `numpy.where`.

Comment (lkluft) [reply easygems/healpix/__init__.py:49]: I don't see how we use `where` to both broadcast the sparse array **and** fill the corresponding values 🤔 But we can also discuss the general approach offline 😉

File: easygems/healpix/__init__.py (hunk 1 above)
Comment (lkluft) [reply easygems/healpix/__init__.py]: Done 👌

File: easygems/healpix/__init__.py (hunk 2 above)
Comment (d70-t) [reply easygems/healpix/__init__.py:49]: Probably that's right 🤔

**Review by d70-t — APPROVED** [review]

**Pull request merged**